test/unit/app/controllers/metamask-controller-test.js

```diff
@@ -816,6 +817,39 @@ describe('MetaMaskController', function () {
 
       await promise
     })
+
+    it('adds a tabId and origin to requests', function (done) {
+      const messageSender = {
+        url: 'http://mycrypto.com',
+        tab: { id: 456 },
+      }
+      streamTest = createThoughStream((chunk, _, cb) => {
+        if (chunk.data && chunk.data.method) {
+          cb(null, chunk)
+        } else {
+          cb()
+        }
+      })
+
+      metamaskController.setupUntrustedCommunication(streamTest, messageSender)
+
+      streamTest.write({
+        name: 'provider',
+        data: {
+          id: 1999133338649204,
+          jsonrpc: '2.0',
+          params: ['mock tx params'],
+          method: 'eth_sendTransaction',
+        },
+      }, null, () => {
+        setTimeout(() => {
+          assert.equal(metamaskController.txController.newUnapprovedTransaction.getCall(0).args[0], 'mock tx params')
```

Comment:
Can we use `calledOnceWithExactly` here to merge this into a single assertion?

Reply:
Might need to rebase this on develop for #7968

Reply:
I've merged this into a single assertion, but I did a `deepStrictEquals` instead, as it gave more informative errors upon failure.